src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -1,131 +1,138 @@
-            ExtendedExpiresOnUnixTimestamp = CoreHelpers.DateTimeToUnixTimestamp(accessTokenExtendedExpiresOn);
-            CachedAt = CoreHelpers.CurrDateTimeInUnixTimestamp();
-            RawClientInfo = rawClientInfo;
-
-            InitUserIdentifier();
-        }
-
-        [DataMember(Name = "realm")]
         internal string TenantId { get; set; }
 
         /// <summary>
         /// String comprised of scopes that have been lowercased and ordered.
         /// </summary>
         /// <remarks>Normalization is important when creating unique keys.</remarks>
-        [DataMember(Name = "target", IsRequired = true)]
         internal string NormalizedScopes { get; set; }
-
-        [DataMember(Name = "cached_at", IsRequired = true)]
         internal string CachedAt { get; set; }
-
-        [DataMember(Name = "expires_on", IsRequired = true)]
         internal string ExpiresOnUnixTimestamp { get; set; }
-
-        [DataMember(Name = "ext_expires_on")]
         internal string ExtendedExpiresOnUnixTimestamp { get; set; }
-
-        [DataMember(Name = "user_assertion_hash", EmitDefaultValue = false)]
         public string UserAssertionHash { get; set; }
 
-        [DataMember(Name = "access_token_type")]
-        internal string TokenType { get; set; }
+        internal string Authority =>
+            string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
 
-        internal string Authority => string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
         internal SortedSet<string> ScopeSet => ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
 
-        internal DateTimeOffset ExpiresOn
+        internal DateTimeOffset ExpiresOn => CoreHelpers.UnixTimestampStringToDateTime(ExpiresOnUnixTimestamp);
+        internal DateTimeOffset ExtendedExpiresOn => CoreHelpers.UnixTimestampStringToDateTime(ExtendedExpiresOnUnixTimestamp);
+        public bool IsExtendedLifeTimeToken { get; set; }
+
+        internal static MsalAccessTokenCacheItem FromJsonString(string json)
+        {
+            return FromJObject(JObject.Parse(json));
+        }
+
+        internal static MsalAccessTokenCacheItem FromJObject(JObject j)
         {
-            get
+            long cachedAt = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.CachedAt);
+            long expiresOn = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExpiresOn);
+
+            // This handles a bug with the name in previous MSAL.  It used "ext_expires_on" instead of
+            // "extended_expires_on" per spec, so this works around that.
+            long ext_expires_on = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExtendedExpiresOn_MsalCompat);
+            long extendedExpiresOn = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExtendedExpiresOn);
+            if (extendedExpiresOn == 0 && ext_expires_on > 0)
```

Comment:
>== 0 [](start = 34, length = 4)

can you explain why one is == 0 and the other is > 0?

Reply:
if the extendedExpiresOn value is not set (default 0) but ext_expires_on is not 0, then use the value in ext_expires_on.  it means that extendedExpiresOn is the default used value, but if it's not set then we use ext_expires_on